### PR TITLE
Feature: Data Preparation

### PR DESCRIPTION
**Pull request by martinctc**
# Summary
This branch adds any data preparation scripts for Google Sheets, prior to feeding into Shiny app. Runs once (not run during deployment).

# Changes
The changes made in this PR are:
1. Added googlesheets_dataclean.R, under .development
1. Added .development to .Rbuildignore

# Check
- [x] All checks pass
- [x] No errors in googlesheets_dataclean.R
- [x] Code in googlesheets_dataclean.R is well documented
- [x] .development is added to .Rbuildignore

### CODE REVIEW

File: .development/googlesheets_dataclean.R [hunk 1]
```diff
@@ -0,0 +1,40 @@
+library(tidyverse)
+library(googlesheets4)
+
+sheet_url <- "https://docs.google.com/spreadsheets/d/1usk9Q-5lA4bL_z6KXpUohc_2x_KhDgLxtm-YEtim_yk/"
+
+# define google sheets ----------------------------------------------------
+## may consider using purrr once we get to more sheets
```
Comment (avisionh) [inline .development/googlesheets_dataclean.R]: `purrr` is beautiful

File: .development/googlesheets_dataclean.R [hunk 2]
```diff
@@ -0,0 +1,40 @@
+library(tidyverse)
```
Comment (avisionh) [inline .development/googlesheets_dataclean.R]: Just a note that adding the whole suite of `tidyverse` packages is a lot of overhead as it's not always necessary to use all the packages in `tidyverse`. It's okay to do here as this sits outside the app.

File: .development/googlesheets_dataclean.R [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+data_master_wanchai <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Wan Chai")
+data_master_eastern <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Eastern")
+data_master_southern <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Southern")
+
+
+# clean googlesheets ------------------------------------------------------
+
+clean_gsheet_dat <- function(data){
+  data %>%
+    janitor::clean_names() %>%
```
Comment (avisionh) [inline .development/googlesheets_dataclean.R]: If you did `clean_names()` instead, would you get the same results?

File: .development/googlesheets_dataclean.R [hunk 4]
```diff
@@ -0,0 +1,40 @@
+library(tidyverse)
+library(googlesheets4)
```
Comment (avisionh) [inline .development/googlesheets_dataclean.R]: Could you do `renv::snapshot()` to record the packages in the *renv.lock* file? Also can you add them to the *DESCRIPTION* file as well?

File: .development/googlesheets_dataclean.R (hunk 3 above)
Comment (avisionh) [reply .development/googlesheets_dataclean.R]: Just curious as to why you need to prepend `clean_names()` since there doesn't seem to be a conflict in multiple functions being called this.

File: .development/googlesheets_dataclean.R [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+# define google sheets ----------------------------------------------------
+## may consider using purrr once we get to more sheets
+
+data_master_cnw <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Central & Western")
+data_master_wanchai <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Wan Chai")
+data_master_eastern <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Eastern")
+data_master_southern <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Southern")
+
+
+# clean googlesheets ------------------------------------------------------
```
Comment (avisionh) [inline .development/googlesheets_dataclean.R:18]: Liking the use of section headers.

File: .development/googlesheets_dataclean.R [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+sheet_url <- "https://docs.google.com/spreadsheets/d/1usk9Q-5lA4bL_z6KXpUohc_2x_KhDgLxtm-YEtim_yk/"
+
+# define google sheets ----------------------------------------------------
+## may consider using purrr once we get to more sheets
+
+data_master_cnw <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Central & Western")
+data_master_wanchai <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Wan Chai")
+data_master_eastern <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Eastern")
+data_master_southern <- googlesheets4::read_sheet(ss = sheet_url, sheet = "Southern")
```
Comment (avisionh) [inline .development/googlesheets_dataclean.R]: If you wanted to use `purrr`, it would be like below:
```suggestion
list_arguments <- list(sheet = c("Central & Western", "Wan Chai", "Eastern", "Southern"))
list_data <- pmap(.l = vec_sheet, .f = read_sheet, ss = sheet_url)
names(list_data) <- paste0("data_master_", c("cnw", "wanchai", "eastern", "southern"))
```

File: .development/googlesheets_dataclean.R [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+
+# bind list and assign to master ------------------------------------------
+
+master_sheet <-
+  list(data_master_cnw,
+       data_master_wanchai,
+       data_master_eastern,
+       data_master_southern) %>%
```
Comment (avisionh) [inline .development/googlesheets_dataclean.R]: If you apply `purrr` above in a way akin to what I suggested [above](https://github.com/avisionh/dashboard-hkdistrictcouncillors/pull/8/files#r452510722), then you can shorten this to:
```suggestion
  list_data %>%
```

**Review by avisionh — CHANGES_REQUESTED** [review]
All passes check but would like `renv::snapshot()` to be run and the packages used in this script added to the *DESCRIPTION* file. 

Rest of my comments are just suggestions if you're interested. :)

File: .development/googlesheets_dataclean.R (hunk 2 above)
Comment (martinctc) [reply .development/googlesheets_dataclean.R]: Yes, the original intention was that this would just sit outside the app so did not take dependencies etc. into consideration. Since we're adding the packages here to renv, I might replace {tidyverse} with individual packages like {dplyr} and {purrr}. What do you think?

Comment (avisionh) [reply .development/googlesheets_dataclean.R]: Yeah now that you mention it, calling the individual packages will make the installing of packages when running `renv::restore()` faster.

File: .development/googlesheets_dataclean.R [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+clean_gsheet_dat <- function(data){
+  data %>%
+    clean_names() %>%
+    separate(col = xuan_qu_constituency, into = c("Constituency_ZH", "Constituency_EN"), sep = "\n", extra = "merge") %>%
+    separate(col = dang_pai_party, into = c("Party_ZH", "Party_EN"), sep = "\n", extra = "merge") %>%
+    separate(col = qu_yi_yuan_dc, into = c("DC_ZH", "DC_ZN"), sep = "\n", extra = "merge") %>%
+    rename(ConstituencyCode = "xuan_qu_hao_ma_constituency_code") %>%
+    mutate(DropDownText = paste0(ConstituencyCode, ": ", Constituency_ZH, " / ", Constituency_EN))
+}
```
Comment (avisionh) [inline .development/googlesheets_dataclean.R]: Should probably hive this off in a folder that contains R scripts for all the functions in the future.

Also, should include unit testing via `testthis` package too.

**Review by avisionh — APPROVED** [review]
Checked the following:
- [x] The code works.
- [x] The packages used are recorded in `renv.lock` and `DESCRIPTION`.
- [x] travis.ci CI/CD and R CMD builds pass.
